### ISSUE TITLE
Remove/rename `aesara.tensor.nnet` sub-package

### ISSUE THREAD

**Issue by twiecki** [aesara-devs/aesara#674]
I think having `nnet` as a name is pretty confusing, these are general math functions not only used for neural nets, and by now Aesara isn't geared towards neural nets anyway. Maybe we can rename (to what?) or, better I think, just move it down to the regular name space.

**Comment by ricardoV94** [comment]
There is a lot of things there that we might not want to move. All those crossentropy / bias variations. 

We already moved sigmoid and softplus out, and I think the next step is to move softmax/log_softmax out (#673).

**Comment by brandonwillard** [comment]
I would go so far as to say that we should disable auto-importing and add a deprecation warning to the `aesara.tensor.nnet` sub-package ASAP, because we don't intend to maintain any of that neural network code as part of core Aesara.

Also, since the tests for that sub-package take a good amount of our CI time (~2 hours), we could subsequently disable testing for it and significantly speed up testing.

**Comment by twiecki** [comment]
That makes sense.

**Comment by twiecki** [comment]
For someone motivated, they could build a NN package on top of aesara, and this might be a good starting point.

**Comment by canyon289** [comment]
> I would go so far as to say that we should disable auto-importing and add a deprecation warning to the `aesara.tensor.nnet` sub-package ASAP, because we don't intend to maintain any of that neural network code as part of core Aesara.
> 
> Also, since the tests for that sub-package take a good amount of our CI time (~2 hours), we could subsequently disable testing for it and significantly speed up testing.

Agree with this, it was a huge slowdown when I was getting developing with aesara both because of all the things that broke in it when I made changes, and the testing slowdown. Removing would streamline codebase tremendously

**Comment by rlouf** [comment]
I agree we should remove all the functionalities in `nnet` that are specific to neural nets in Aesara (and `signal`?).

When I have time I'd like to move these functionalities to a Keras-like library that implements the layers as `OpFromGraph`s to demonstrate what Aesara can do.

**Comment by rth** [comment]
So from which module should one import for instance the convolution operator once that sub-package is indeed removed?

**Comment by brandonwillard** [comment]
> So from which module should one import for instance the convolution operator once that sub-package is indeed removed?

We have discussed moving the `nnet` material to another package, but, since `conv` is a SciPy-like feature, we could just as well keep it with the existing SciPy support in Aesara.  We won't remove it until that next step is clear.

Regardless, if there are people still depending on this code, we will at the very least keep it available in some form before removing it from here.

**Comment by senarvi** [comment]
@brandonwillard I have an old Theano project and wanted to see if I can run it with Aesara. The only thing that didn't work out of the box is the convolution operations. Am I correct assuming that the nnet module is broken and not maintained anymore? The error I get is

`aesara.graph.rewriting.basic.MetaNodeRewriterSkip: AbstractConv2d Aesara rewriting failed: there is no implementation available supporting the requested options. If on CPU, do you have a BLAS library installed Aesara can link against? On the CPU we do not support float16.`

Installing g++, BLAS, SciPy, and JAX didn't help.
